api/src/test/java/org/openmrs/api/impl/ConditionServiceImplTest.java

```diff
@@ -0,0 +1,172 @@
+		Condition firstActiveCondition = activeConditions.get(0);
+		Condition secondActiveCondition = activeConditions.get(1);
+		String firstUUID = "2cc6880e-2c46-11e4-9138-a6c5e4d20fb7";
+		String secondUUID = "2cc6880e-2c46-15e4-9038-a6c5e4d22fb7";
+		Assert.assertTrue(activeConditions.size() == 2);
+		Assert.assertEquals(firstUUID,firstActiveCondition.getUuid());
+		Assert.assertEquals(secondUUID, secondActiveCondition.getUuid());
+	}
+	
+	
+
+	/**
+	 * @see ConditionService#voidCondition(Condition, String) 
+	 */
+	@Test
+	public void voidCondition_shouldVoidConditionSuccessfully(){
+		String voidReason = "Test Reason";
+		Condition nonVoidedCondition = conditionService.getCondition(2);
+		Assert.assertFalse(nonVoidedCondition.getVoided());
+		Assert.assertNull(nonVoidedCondition.getVoidReason());
+		Assert.assertNull(nonVoidedCondition.getDateVoided());
+		Assert.assertNull(nonVoidedCondition.getVoidedBy());
+		
+		Condition voidedCondition = conditionService.voidCondition(nonVoidedCondition, voidReason);
+		Assert.assertEquals(ConditionVerificationStatus.CONFIRMED, voidedCondition.getVerificationStatus());
+		Assert.assertEquals(ConditionClinicalStatus.ACTIVE, voidedCondition.getClinicalStatus());
+		Assert.assertEquals("2cc6880e-2c46-11e4-9138-a6c5e4d20fb7", voidedCondition.getUuid());
+		Assert.assertTrue(voidedCondition.getVoided());
+		Assert.assertEquals(voidReason, voidedCondition.getVoidReason());
+		Assert.assertNotNull(voidedCondition.getDateVoided());
+		Assert.assertEquals(Context.getAuthenticatedUser(), voidedCondition.getVoidedBy());
+	}
+
+	/**
+	 * @see ConditionService#unvoidCondition(Condition) 
+	 */
+	@Test
+	public void unvoidCondition_shouldUnvoidConditionSuccessfully(){
+		Condition voidedCondition = conditionService.voidCondition(conditionService.getCondition(2), "Test Reason");
+		Assert.assertTrue(voidedCondition.getVoided());
+		Assert.assertNotNull(voidedCondition.getVoidReason());
+		Assert.assertNotNull(voidedCondition.getDateVoided());
+		Assert.assertEquals(Context.getAuthenticatedUser(), voidedCondition.getVoidedBy());
+		
+		Condition unVoidedCondition = conditionService.unvoidCondition(voidedCondition);
+		Assert.assertEquals(ConditionVerificationStatus.CONFIRMED, unVoidedCondition.getVerificationStatus());
+		Assert.assertEquals(ConditionClinicalStatus.ACTIVE, unVoidedCondition.getClinicalStatus());
+		Assert.assertEquals("2cc6880e-2c46-11e4-9138-a6c5e4d20fb7", unVoidedCondition.getUuid());
+		Assert.assertFalse(unVoidedCondition.getVoided());
+		Assert.assertNull(unVoidedCondition.getVoidReason());
+		Assert.assertNull(unVoidedCondition.getDateVoided());
+		Assert.assertNull(unVoidedCondition.getVoidedBy());
+	}
+
+	/**
+	 * @see ConditionService#purgeCondition(Condition) 
+	 */
+	@Test
+	public void purgeCondition_shouldPurgeCondition() {
+		Condition existingCondition = conditionService.getCondition(1);
```

Comment:
Since you are using 1 in more than one place in this method, then assign it to a constant variable.

Reply:
Okay